addon/object/index.js

```diff
@@ -179,7 +191,17 @@ export const computed = decoratorWithParams(function(target, key, desc, params)
 
     // Use a standard ember computed since getter/setter arrity is restricted,
     // meaning ember-macro-helpers doesn't provide any benefit
-    return emberComputed(...params, { get, set: setter });
+    if (SUPPORTS_NEW_COMPUTED) {
+      return emberComputed(...params, { get, set: setter });
+    } else {
+      return emberComputed(...params, function() {
```

Comment:
I don't think this will work, will it? You need to make the function arity > 1 or Ember won't use the setter (and instead will clobber), right? Am I completely mis-remembering this?